magento-research/pwa-studio
# Refactor driver usage to improve Venia portability

_This PR is a cherry-pick from #1198 to make that PR more digestible._

It is number 3 of 6 in the chain of cherry-picked PRs. It needs to be merged before the next 3:

4. #1219 
5. #1224
6. #1198

## Description
When trying to use Venia components in a context with another app, it's easy to get React errors claiming you're trying to use a `Link` element without a `Router` in context. This mostly happens because we forgot a couple of spots where we are hardcoding a particular instance of React Router instead of pulling in the instance provided by `@magento/venia-drivers`. This PR does the refactoring and prop drilling necessary to pass drivers around, so that an app can share routing with other components.

## Related Issue





Closes #ISSUE_NUMBER.

## Verification Steps

1. Ensure `yarn build` is successful
1. Run `yarn watch:venia` and load up your local Venia in your browser.
1. Ensure the main (CMS) page works.
1. Click through to the Create Account page, ensure that works.
1. Execute a search and ensure that works.
1. Hit a page that doesn't exist (ex: `/noexist.html`) and ensure you see the "Something went wrong. Please try again" message.
1. Sign In and ensure you can pull up your account by clicking on your user information:

![Screen Shot 2019-05-09 at 10 13 16 AM](https://user-images.githubusercontent.com/13182778/57464873-20ca1280-7243-11e9-9b4b-d5925f1225d5.png)

## How Have YOU Tested this?




## Screenshots / Screen Captures (if appropriate):

## Proposed Labels for Change Type/Package




## Checklist:
